src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1085,4 +1089,140 @@ BOOST_AUTO_TEST_CASE(CustomRecursiveReberGrammarTest)
   ReberGrammarTestCustomNetwork(16, true);
 }
 
+/**
+ * @brief Generates noisy sine wave.
+ *
+ * @param data    The data points as output
+ * @param labels  The expected values as output
+ * @param rho     The size of the sequence of each data point
+ * @param outputSteps How many output steps to consider
+ * @param dataPoints  The number of input data points. The actual generated data
+ * points may be more than this to adjust to the outputSteps. But at the minimum
+ * these many data points will be generated.
+ * @param gain    The gain on the amplitude
+ * @param freq    The frquency of the sine wave
+ * @param phase   The phase shift if any
+ * @param noisePercent  The percent noise to induce
+ * @param numCycles  How many full size wave cycles required. All the data
+ * points will be fit into these cycles.
+ * @param normalize Whether to normalise the data. This may be required for some
+ * layers like LSTM. Default is true.
+ */
+void GenerateNoisySinRNN(arma::cube& data, arma::cube& labels, size_t rho,
```

Comment:
I think maybe we should rename this---it's not creating an `RNN`, it's creating data, so I think the method name should reflect that.

Reply:
Well you are right on the data. At the end though I rearrange the data to make two cubes of input data and labels so that they can be used in the RNN training directly. Hence the name. I changed the documentation on the code also. Let me think of a better name..